src/RoatpProviderModeration/SFA.DAS.RoatpProviderModeration.Application/Queries/GetProvider/GetProviderQueryHandler.cs

```diff
@@ -0,0 +1,43 @@
+﻿using MediatR;
+using Microsoft.Extensions.Logging;
+using SFA.DAS.RoatpProviderModeration.Application.InnerApi.Requests;
+using SFA.DAS.RoatpProviderModeration.Application.InnerApi.Responses;
+using SFA.DAS.SharedOuterApi.Configuration;
+using SFA.DAS.SharedOuterApi.Interfaces;
+using System.Net;
+
+namespace SFA.DAS.RoatpProviderModeration.Application.Queries.GetProvider
+{
+    public class GetProviderQueryHandler : IRequestHandler<GetProviderQuery, GetProviderQueryResult>
+    {
+        private readonly IRoatpCourseManagementApiClient<RoatpV2ApiConfiguration> _courseManagementApiClient;
+        private readonly ILogger<GetProviderQueryHandler> _logger;
+
+        public GetProviderQueryHandler(IRoatpCourseManagementApiClient<RoatpV2ApiConfiguration> courseManagementApiClient, ILogger<GetProviderQueryHandler> logger)
+        {
+            _courseManagementApiClient = courseManagementApiClient;
+            _logger = logger;
+        }
+
+        public async Task<GetProviderQueryResult> Handle(GetProviderQuery request, CancellationToken cancellationToken)
+        {
+            _logger.LogInformation("Get Provider request received for ukprn {ukprn}", request.Ukprn);
+
+            var response = await _courseManagementApiClient.GetWithResponseCode<GetProviderResponse>(new GetProviderRequest(request.Ukprn));
+
+            GetProviderQueryResult provider; 
+
+            if (response.StatusCode == HttpStatusCode.OK) provider = response?.Body;
```

Comment:
You can avoid else's by returning early. If you want to stick to this pattern, add braces to each condition to change the statements to blocks.